allennlp_models/rc/models/transformer_qa.py

```diff
@@ -247,9 +249,30 @@ def _collect_best_span_strings(
         """
         _best_spans = best_spans.detach().cpu().numpy()
 
+        min_node_batch_size = len(metadata)
+        if is_distributed():
+            # In the distributed case, if the batch isn't balanced across nodes, we'll get a dead-lock
+            # if one node has more instances in a batch than another node.
+            if dist.get_backend() == "nccl":
+                device = torch.cuda.current_device()
+            else:
+                device = torch.device("cpu")
+            _min_node_batch_size = torch.tensor(min_node_batch_size, dtype=torch.int, device=device)
+            dist.all_reduce(_min_node_batch_size, op=dist.ReduceOp.MIN)
+            min_node_batch_size = _min_node_batch_size.item()
```

Comment:
This is becoming a really common pattern. It would be to have a helper function for this called `dist_reduce` or something. Then we could just do:

```suggestion
            min_node_batch_size = dist_reduce(min_node_batch_size, dist.ReduceOp.MIN, dtype=torch.int)
```

Reply:
Sounds like a good change.